origin-js/src/resources/marketplace.js

```diff
@@ -108,25 +109,36 @@ class Marketplace {
     const ipfsHash = this.contractService.getIpfsHashFromBytes32(
       chainOffer.ipfsHash
     )
+
     const ipfsOffer = await this.ipfsDataStore.load(OFFER_DATA_TYPE, ipfsHash)
 
     // validate offers awaiting approval
     if (chainOffer.status === 'created') {
       const listing = await this.getListing(listingId)
+      const listingCommision =
+          listing.commission && typeof listing.commission === 'object' ?
+            await this.contractService.moneyToUnits(listing.commission) :
+            '0'
+
+      if (listing.type === 'unit') {
+        // TODO(John) - there is currently no way to know the currency of a fractional listing.
+        // We probably need to add a required "currency" field to the listing schema and write a check here
+        // to make sure the chainOffer and the listing have the same currency
+        const listingCurrency = listing.price && listing.price.currency
+        const listingPrice = await this.contractService.moneyToUnits(listing.price)
+        const listingPriceNum = parseFloat(listingPrice)
```

Comment:
Sorry to be such a pita... isn't there a risk of overflow since these could be really large numbers because they are in wei/natural units ?

How about using BigNumber rather than float ? Then we can use the BigNumber isGreaterThan method to do the comparison below. Actually perhaps no need for intermediary variables listingPriceNum and chainOfferValueNum. line #139 could be changed to something like
```
if (BigNumber(listingPrice).isGreaterThan(BigNumber(chainOffer.value))) {
 throw new Error('Invalid offer: insufficient offer amount for listing')
}
```

And how about doing the same for the commission comparison at line #144 ?

I actually wonder why our unit tests did not catch this ?